+toml/struct_to_map.m

```diff
@@ -0,0 +1,24 @@
+% STRUCT_TO_MAP convert a struct to a (possibly nested) containers.Map object
+%
+%   STRUCT_TO_MAP(my_struct) returns an ordered map with the same keys and corresponding
+%   values as `my_struct`, recursively converting struct values likewise. Any keys
+%   that are not already valid MATLAB identifiers will be preserved as is.
+
+function map = struct_to_map(struct)
+    if isstruct(struct)
+        map = containers.Map();
+        fields = fieldnames(struct);
+        for i = 1 : numel(fields)
+            key = fields{i};
+            value = NewPreferences.struct_to_map(struct.(key));
```

Comment:
Can you change the module name to `toml` here?

```suggestion
            value = toml.struct_to_map(struct.(key));
```

Reply:
Yea of course, the updated code has now been committed.